### PR TITLE
Update losses to include MSE, Loss base class, track individual components of weighted sum

### PR DESCRIPTION
**Pull request by dhpitt**

### CODE REVIEW

File: neuralop/losses/meta_losses.py [hunk 1]
```diff
@@ -62,28 +67,39 @@ def __call__(self, pred: torch.Tensor, truth: torch.Tensor, **kwargs):
...
-    def __init__(self, losses, weights=None):
+    def __init__(self, losses, weights=None, return_individual=True, compute_grads=False):
         super().__init__()
         if weights is None:
             weights = [1.0 / len(losses)] * len(losses)
         if not len(weights) == len(losses):
             raise ValueError("Each loss must have a weight.")
-        self.losses = list(zip(losses, weights))
+        #self.losses = list(zip(losses, weights))
+        self.losses = {x.__name__: [x,y] for x,y in zip(losses,weights)}
```
Comment (juliusberner) [inline neuralop/losses/meta_losses.py:82]: why `__name__` instead of `name`?

File: neuralop/losses/loss.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+"""
+Base Loss class for neuralop
+"""
+from  abc import ABC, abstractmethod
+from torch import nn
+
+class Loss(nn.Module, ABC):
```
Comment (juliusberner) [inline neuralop/losses/loss.py:7]: Why does `Loss` need to inherit from `nn.Module`?

File: neuralop/training/trainer.py [hunk 3]
```diff
@@ -197,16 +216,32 @@ def train(
...
-                if self.overrides_loss:
+                if self.overrides_training_loss:
                     loss += self.callbacks.compute_training_loss(
                         out=out, **sample, amp_autocast=self.amp_autocast
                     )
                 else:
-                    if self.amp_autocast:
-                        with amp.autocast(enabled=True):
-                            loss += training_loss(out, **sample)
+                    if self.monitor_individual_train_losses:
```
Comment (juliusberner) [inline neuralop/training/trainer.py:224]: An idea would be to always output a loss dictionary (i.e., automatically use the weighted loss, even for a single loss) to simplify the trainer?

File: neuralop/training/trainer.py [hunk 4]
```diff
@@ -197,16 +216,32 @@ def train(
...
+                            with amp.autocast(enabled=True):
+                                loss_outputs = training_loss(out, **sample)
+                        else:
+                            loss_outputs = training_loss(out, **sample)
+
+                        for loss_name, loss_value in loss_outputs.items():
+                            loss += loss_value
+                            train_losses[loss_name] += loss_value
+                            # accumulate training gradients for each subloss over the epoch
+                            if self.compute_train_grads:
```
Comment (juliusberner) [inline neuralop/training/trainer.py:235]: This is a nice feature, but we typically do not want to do that in every train step but just periodically.

Comment (juliusberner) [reply neuralop/training/trainer.py:235]: Theoretically, we could also save one grad computation (i.e., the `.backward`) by aggregating the grads appropriately, but I think it is not necessary if we only compute it periodically.

File: neuralop/training/trainer.py [hunk 5]
```diff
@@ -142,6 +143,15 @@ def train(
 
         if training_loss is None:
             training_loss = LpLoss(d=2)
```
Comment (juliusberner) [inline neuralop/training/trainer.py:145]: Should we set the default to `MSE`?

Comment (JeanKossaifi) [reply neuralop/training/trainer.py:145]: Default Lp loss should be MSE (l2 loss)

**Comment by dhpitt** [comment]
Closing to split this into 2 separate PRs. MSE is simpler and more urgent
